service/account/internal/src/main/java/org/eclipse/kapua/service/account/internal/AccountImpl.java

```diff
@@ -178,5 +182,22 @@ public void setExpirationDate(Date expirationDate) {
         this.expirationDate = expirationDate;
     }
 
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (!(object instanceof Account)) {
+            return false;
+        }
 
+        final Account that = (Account) object;
+
+        return getId() != null ? getId().equals(that.getId()) : that.getId() == null;
```

Comment:
I don't think that just comparing the id is sufficient to determine if two Account objects are equal; it should be comparing all of the fields.  For example, if you are updating the expiration date for an Account, it should no longer be considered "equals" to the original object with the same id but different expiration date.

I realize that this can get complicated dealing with parent classes and other entity types, so I'd be okay proceeding with this and dealing with proper "equals" and "hashCode" implementations as a separate issue.

Reply:
Considering that the class AccountImpl is the JPA representation of the database table, it is common to reflect the primary key of the table in the equals/hashcode. The kind of comparison you are referring to often requires custom logic, and is better left in an external collaborator. The intended use here is to check whether we are talking about the same entity in a broader sense. 
That said, given the other changes introduced here and elsewhere, this particualar addition of equals and hashcode could even be omitted.